From 4a56b7af6f8c33aff1e41a8932d111b46615b9e0 Mon Sep 17 00:00:00 2001
From: Tim Snyder <snyder.tim@gmail.com>
Date: Sat, 9 Apr 2022 15:20:00 +0000
Subject: [PATCH] add --enable_libdwarf and --enable_dwarfdump to configure

By disabling libdwarf and dwarfdump with configure we can use the tests
against already installed libdwarf and dwarfdump.
---
 Makefile.am  | 12 +++++++++++-
 configure.ac | 29 +++++++++++++++++++++++++++--
 2 files changed, 38 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 83c14e979..0ed335a40 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,7 +3,17 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = libdwarf dwarfdump
+AUTOMAKE_OPTIONS=subdir-objects
+SUBDIRS =
+
+if HAVE_LIBDWARF
+SUBDIRS += libdwarf
+endif
+
+if HAVE_DWARFDUMP
+SUBDIRS += dwarfdump
+endif
+
 if HAVE_DWARFGEN
 SUBDIRS += dwarfgen
 endif
diff --git a/configure.ac b/configure.ac
index 0e84d64da..734d52e26 100644
--- a/configure.ac
+++ b/configure.ac
@@ -28,6 +28,31 @@ AC_CONFIG_MACRO_DIRS([m4])
 
 ### Additional options to configure
 
+AC_ARG_ENABLE([libdwarf],
+   [AS_HELP_STRING([--enable-libdwarf],
+                   [enable libdwarf compilation @<:@default=yes@:>@])],
+   [
+    AS_IF(
+        [test "x${enableval}" = "xyes"],
+        [enable_libdwarf="yes" ],
+        [enable_libdwarf="no"])
+   ],
+   [enable_libdwarf="yes"])
+AM_CONDITIONAL([HAVE_LIBDWARF], [test "x${enable_libdwarf}" = "xyes"])
+
+AC_ARG_ENABLE([dwarfdump],
+   [AS_HELP_STRING([--enable-dwarfdump],
+                   [enable dwarfdump compilation @<:@default=yes@:>@])],
+   [
+    AS_IF(
+        [test "x${enableval}" = "xyes"],
+        [enable_dwarfdump="yes" ],
+        [enable_dwarfdump="no"])
+   ],
+   [enable_dwarfdump="yes"])
+AM_CONDITIONAL([HAVE_DWARFDUMP], [test "x${enable_dwarfdump}" = "xyes"])
+
+
 AC_ARG_ENABLE([dwarfgen],
    [AS_HELP_STRING([--enable-dwarfgen],
                    [enable dwarfgen compilation @<:@default=no@:>@])],
@@ -755,7 +780,7 @@ echo
 echo "  zlib support.........: ${have_zlib}"
 echo "  sanitize support.....: ${enable_sanitize}"
 echo
-echo "  libdwarf.............: always"
+echo "  libdwarf.............: ${enable_libdwarf}"
 echo "    old frame column...: ${enable_oldframecol}"
 echo "    names table........: ${enable_namestable}"
 echo "    elf64_getehdr......: ${have_getehdr}"
@@ -765,7 +790,7 @@ echo "    Elf64_Sym..........: ${have_elf64_sym}"
 echo "    Elf spelled........: ${struct_elf}"
 echo "    Windows path corr..: ${enable_windowspath}"
 echo "    Nonstandardprintf..: ${enable_nonstandardprintf}"
-echo "  dwarfdump............: always"
+echo "  dwarfdump............: ${enable_dwarfdump}"
 echo "    elf64_getehdr......: ${have_getehdr}"
 echo "    Elf64_Rel (r_info).: ${have_elf64_rel}"
 echo "    regex..............: ${have_regex}"
-- 
1.8.3.1

